src/lang-items.md

```diff
@@ -197,6 +159,14 @@ the screen. While the language item's name is `panic_impl`, the symbol name is
 Finally, a `eh_catch_typeinfo` static is needed for certain targets which
 implement Rust panics on top of C++ exceptions.
 
+## Well-known paths
+
+In several cases compiler finds specific item not by `lang` attribute. Instead
+item path is hardcored. For example compiler assumes `Iterator` trait to be
+available as `core::iter::Iterator`. This only happens when item is required
+on early compilation stages (for example `Iterator` is used in for loops
+desugaring).
+
```

Comment:
Actually, based on [looking at the output of `rustc +nightly -Z unpretty=hir`](https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler/topic/why.20are.20.60Some.60.20and.20.60None.60.20language.20items.3F/near/219286210), `core::iter::Iterator` is *not* hardcoded when desugaring `for` loops. Do you know of a place where the language item's path *is* hardcoded, or should we just remove this section?

Reply:
afaik we now try to use `QPath::LangItem` everywhere instead of using fully qualified paths.

https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/hir/enum.QPath.html#variant.LangItem

I think this is probably out of date

Reply:
Yeah, you're right, the compiler no longer hardcodes paths. IIRC it was true some time ago, so the compiler was improved!

I'll verify if there are any well-known paths. If no I'll delete this section. Thanks!